model/riscv_pte.sail

```diff
@@ -29,6 +29,11 @@ bitfield PTE_Bits : pteAttribs = {
   V : 0
 }
 
+bitfield PTE_Ext : extPte = {
```

Comment:
If you're making this defined by the standard then this should be a separate type (logically, part of pteAttribs, but it may be easier to have pteAttribsHi and pteAttribsLo), not extPte. extPte is for non-standard extensions to the model to use (i.e. just sail-cheri-riscv, currently), but this also needs to be done in a way that those extensions continue to be able to work (code changes are fine and may well be necessary, but there must be a way for them to hook into sail-riscv as needed without modifying sail-riscv).

Reply:
Thanks for the feedback @jrtc27.  Is renaming to pteAttribsHi and pteAttribsLo enough to avoid that conflict, or are you suggesting something else also needs to be done?

Reply:
Yes and no. It means `extPte` is free to be defined by extensions again, but there's still the issue of changing what you can do with it; either it overlaps with some/all of pteAttribsHi and things get really weird, or you reduce it to the bits not defined by the standard and then extensions using them break. Either way there are issues for CHERI-RISC-V; the former means composing Zsn and Xcheri doesn't work (but if done correctly you should still be able to support them as mutually exclusive extensions in the model), the latter just breaks it entirely and forces us to change CheriBSD, our four/five different hardware implementations, QEMU, spike and, of course, sail-cheri-riscv, to pick different bits for the 5 bits (and researchers at Microsoft are reserving a 6th bit) we need (and then rinse and repeat as RISC-V gobbles more up in the coming years), which is manageable for a research group but extremely painful when we have downstreams at other universities and companies across the world building things on CHERI-RISC-V. There was no reserved space for non-standard extensions so we picked what we thought was most likely to remain unused (yeah, picking reserved bits not marked for that is a risk, but we had no other option), but here we are. So I don't know what the right answer is beyond being able to convince the virtual memory task group to leave those bits reserved for non-standard extensions.

Reply:
Also, this was done based on the fact that riscv-privileged says "These reserved bits may also be used to facilitate research experimentation.", which is exactly what we're doing.